invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/errors/FormFeedback.js

```diff
@@ -134,122 +111,76 @@ class DisconnectedFormFeedback extends Component {
-    });
-    const pids = errors.pids || {};
-    const step0Pids = _isObject(pids)
-      ? Object.entries(pids).map(([key, value]) => {
-          return ["pids." + key, value];
-        })
-      : [["pids", pids]];
-    const customFields = errors.custom_fields || {};
-    const step0CustomFields = Object.entries(customFields).map(([key, value]) => {
-      return ["custom_fields." + key, value];
+      // Try to match error to a section based on field paths
+      for (const [section, fields] of Object.entries(this.sections)) {
+        if (fields.some((field) => path.startsWith(field))) {
+          const sectionElement = document.getElementById(section);
+          const label = sectionElement?.getAttribute("label") || "Unknown section";
+          errorSections.set(section, {
+            label,
+            count: (errorSections.get(section)?.count || 0) + errorCount,
+          });
+          return;
+        }
+      }
+
+      // If not matched in predefined sections, check dynamically in accordions
+      const sectionElement = document
+        .querySelector(`label[for^="${path.replace(/^(.*?)(\[\d+\].*)?$/, "$1")}"]`)
+        ?.closest(".accordion");
+      if (sectionElement) {
+        const sectionId = sectionElement.id;
+        const label = sectionElement.getAttribute("label") || "Unknown section";
+        errorSections.set(sectionId, {
+          label,
+          count: (errorSections.get(sectionId)?.count || 0) + errorCount,
+        });
+      }
     });
-    const step0 = Object.fromEntries(
-      step0Metadata
-        .concat(step0Files)
-        .concat(step0Access)
-        .concat(step0Pids)
-        .concat(step0CustomFields)
-    );
 
-    // Step 1 - Transform each error value into array of error messages
-    const step1 = Object.fromEntries(
-      Object.entries(step0).map(([key, value]) => {
-        return [key, this.toErrorMessages(value)];
-      })
+    // Order sections based on their DOM appearance and return error links
+    const accordions = Array.from(document.querySelectorAll(".accordion")).map(
+      (accordion) => accordion.id
     );
 
-    // Step 2 - Group error messages by label
-    // (different error keys can map to same label e.g. title and
-    // additional_titles)
-    const labelledErrorMessages = {};
-    for (const key in step1) {
-      const label = this.labels[key] || "Unknown field";
-      let messages = labelledErrorMessages[label] || [];
-      labelledErrorMessages[label] = messages.concat(step1[key]);
-    }
+    const orderedSections = [
+      ...accordions.filter((id) => errorSections.has(id)), // Keep sections that exist in order
+      ...[...errorSections.keys()].filter((id) => !accordions.includes(id)), // Append missing sections
+    ];
 
-    return labelledErrorMessages;
+    return orderedSections.map((sectionId, i) => {
+      const { label, count } = errorSections.get(sectionId);
+      return (
+        <a key={i} className="pl-5" href={`#${sectionId}`}>
```

Comment:
please fix the eslint warning.
the reason why it flags it, is that using iterators in arrays is sometimes messing with the rendering of specific elements - it does not know which ones are really updated as components. Can you use section IDs as keys instead?